src/test/HostActivationTests/NativeHostApis.cs

```diff
@@ -200,15 +200,18 @@ public void Hostfxr_get_available_sdks_with_multilevel_lookup()
                 Path.Combine(f.SelfRegisteredGlobalSdkDir, "15.1.4-preview"),
             });
 
-            f.Dotnet.Exec(f.AppDll, new[] { "hostfxr_get_available_sdks", f.ExeDir })
-                .EnvironmentVariable("TEST_MULTILEVEL_LOOKUP_PROGRAM_FILES", f.ProgramFiles)
-                .EnvironmentVariable("TEST_MULTILEVEL_LOOKUP_SELF_REGISTERED", f.SelfRegistered)
-                .CaptureStdOut()
-                .CaptureStdErr()
-                .Execute()
-                .Should().Pass()
-                .And.HaveStdOutContaining("hostfxr_get_available_sdks:Success")
-                .And.HaveStdOutContaining($"hostfxr_get_available_sdks sdks:[{expectedList}]");
+            using (TestOnlyProductBehavior.Enable(f.Dotnet.GreatestVersionHostFxrFilePath))
```

Comment:
This is going to change the product binary that is in obj (unlike the other tests which I think are all operating on copied versions)? I know the `TestFileBackup` should put it back to the original, but the tests messing with it from there directly seems off to me.

Reply:
Good catch - I modified the tests t use dotnet from the tests folder instead.